### PR TITLE
Simplify embeddings class to nn.Embedding

### PR DESCRIPTION
**Pull request by xloem**
The model does not appear to use position and token type embeddings, nor dropout or layernorm after them. It could be simplifying to remove `Rwkv4NeoEmbeddings` and replace with a simple `nn.Embedding`, optionally migrating `if inputs_embeds is None:` into the encoder class, so the user can provide them manually.

### CODE REVIEW

**Comment by xloem** [comment]
The dev arch actually supports position embeddings in some places. Closing this as a mistake.
